mosviz/loaders/utils.py

```diff
@@ -1,15 +1,51 @@
-from glue.config import data_factory
+import re
 
+from glue.config import data_factory
 
 __all__ = ['mosviz_cutout_loader', 'mosviz_spectrum2d_loader',
-           'mosviz_spectrum1d_loader', 'mosviz_level2_loader']
+           'mosviz_spectrum1d_loader', 'mosviz_level2_loader',
+           'split_file_name']
 
 SPECTRUM1D_LOADERS = {}
 SPECTRUM2D_LOADERS = {}
 CUTOUT_LOADERS = {}
 LEVEL2_LOADERS = {}
 
 
+def split_file_name(file_name):
+    """
+    Used to split fits extension from file name.
+    Extensions are appended to the file path in
+    square brackets as follows:
+        <file_path>[<extension>]
+    If file_name doesn't have an extension,
+    the function will set ext to 0 by default.
+
+    Parameters
+    ----------
+    file_name : str
+        File name to parse
+
+    Returns
+    -------
+    path_and_ext : list
+
+    """
+
+    # Create a regular expression to match the expected pattern
+    # (this should be defined outside the function)
+    regex = re.compile(r'^(.+)\[(\d+)\]$')
```

Comment:
This line should be moved to module-level scope and probably redefined as `FILENAME_REGEX` or something like that. Also the comment above should be deleted.